File: rb/features.rb
# Musicbrainz SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module MusicbrainzFeatures
  def self.make_feature(name)
    case name
    when "base"
      MusicbrainzBaseFeature.new
    when "ratelimit"
      MusicbrainzRatelimitFeature.new
    when "retry"
      MusicbrainzRetryFeature.new
    when "test"
      MusicbrainzTestFeature.new
    when "timeout"
      MusicbrainzTimeoutFeature.new
    else
      MusicbrainzBaseFeature.new
    end
  end
end

File: py/features.py
# Musicbrainz SDK feature factory

from feature.base_feature import MusicbrainzBaseFeature
from feature.test_feature import MusicbrainzTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MusicbrainzBaseFeature(),
        "test": lambda: MusicbrainzTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
